pkg/logger/logger.go

```diff
@@ -0,0 +1,80 @@
+package logger
+
+import (
+	"fmt"
+	"io"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	log *zap.Logger
+)
+
+type Logger struct {
+	Debug bool
+	Quiet bool
+}
+
+func LogLevel(logger Logger) {
+	switch {
+	case logger.Debug:
+		New(zapcore.DebugLevel)
+	case logger.Quiet:
+		New(zapcore.ErrorLevel)
+	default:
+		New(zapcore.InfoLevel)
+	}
+}
+
+func New(level zapcore.Level) {
+	var err error
+
+	config := zap.NewProductionConfig()
+
+	logLevel := zap.NewAtomicLevel()
+	logLevel.SetLevel(level)
+	config.Level = logLevel
+
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.TimeKey = "timestamp"
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	encoderConfig.StacktraceKey = ""
+
+	config.EncoderConfig = encoderConfig
+	config.Encoding = "console"
+	config.OutputPaths = []string{"stdout"}
+	config.ErrorOutputPaths = []string{"stderr"}
+	config.EncoderConfig.EncodeLevel = zapcore.CapitalColorLevelEncoder
+
+	log, err = config.Build(zap.AddCallerSkip(1))
+
+	if err != nil {
+		panic(err)
+	}
+}
+
+// FInfo I need to check if the debug is false because the error comes in the debug also as true
+func FInfo(w io.Writer, message string) {
+	if !(log.Core().Enabled(zapcore.ErrorLevel) && !log.Core().Enabled(zapcore.DebugLevel)) ||
```

Comment:
What's the difference of doing this vs just using fmt.Fprintf directly in commands?

Reply:
Gerenciar o log e manter o comportamento anterior